cli/azd/pkg/project/service_target_aks.go

```diff
@@ -258,23 +288,95 @@ func (t *aksTarget) deployManifests(
 	return deployment, nil
 }
 
+// deployKustomize deploys kustomize manifests to the k8s cluster
+func (t *aksTarget) deployKustomize(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	task *async.TaskContextWithProgress[*ServiceDeployResult, ServiceProgress],
+) (bool, error) {
+	if serviceConfig.K8s.Kustomize == nil {
+		return false, nil
+	}
+
+	task.SetProgress(NewServiceProgress("Applying k8s manifests"))
+	overlayPath, err := serviceConfig.K8s.Kustomize.Directory.Envsubst(t.env.Getenv)
+	if err != nil {
+		return false, fmt.Errorf("failed to envsubst kustomize directory: %w", err)
+	}
+
+	// When deploying with kustomize we need to specify the full path to the kustomize directory.
+	// This can either be a base or overlay directory but must contain a kustomization.yaml file
+	kustomizeDir := filepath.Join(serviceConfig.Project.Path, serviceConfig.RelativePath, overlayPath)
+	if _, err := os.Stat(kustomizeDir); os.IsNotExist(err) {
+		return false, fmt.Errorf("kustomize directory '%s' does not exist: %w", kustomizeDir, err)
+	}
+
+	// Kustomize does not have a built in way to specify environment variables
+	// A common well-known solution is to use the kustomize configMapGenerator within your kustomization.yaml
+	// and then generate a .env file that can be used to generate config maps
+	// azd can help here to create an .env file from the map specified within azure.yaml kustomize config section
+	if len(serviceConfig.K8s.Kustomize.Env) > 0 {
+		builder := strings.Builder{}
+		for key, exp := range serviceConfig.K8s.Kustomize.Env {
+			value, err := exp.Envsubst(t.env.Getenv)
+			if err != nil {
+				return false, fmt.Errorf("failed to envsubst kustomize env: %w", err)
+			}
+
+			builder.WriteString(fmt.Sprintf("%s=%s\n", key, value))
+		}
+
+		// We are manually writing the .env file since k8s config maps expect unquoted values
+		// The godotenv library will quote values when writing the file without an option to disable
+		envFilePath := filepath.Join(kustomizeDir, ".env")
+		if err := os.WriteFile(envFilePath, []byte(builder.String()), osutil.PermissionFile); err != nil {
+			return false, fmt.Errorf("failed to write kustomize .env: %w", err)
+		}
+
+		defer os.Remove(envFilePath)
+	}
+
+	// Another common scenario is to use the kustomize edit commands to modify the kustomization.yaml
+	// configuration before	applying the manifests.
+	// Common scenarios for this would be for modifying the images or namespace used for the deployment
+	for _, edit := range serviceConfig.K8s.Kustomize.Edits {
```

Comment:
Should we be adding the `kustomize` CLI to the list of required tools when `Kustomize.Edits` is not empty so we get our genialized "you need this tool but you don't have it" behavior?